lib/ts/fetch.ts

```diff
@@ -567,167 +595,201 @@ export async function onInvalidClaimResponse(response: ResponseWithBody) {
+            }`
+        );
+    } else {
+        return CookieHandlerReference.getReferenceOrThrow().cookieHandler.setCookie(
+            `${name}=${value};expires=${expires};domain=${domain};path=/;samesite=${
+                AuthHttpRequest.config.isInIframe ? "none;secure" : "lax"
+            }`
+        );
+    }
+}
+
+export async function getTokenForHeaderAuth(tokenType: TokenType) {
+    const name = getStorageNameForToken(tokenType);
+
+    return getFromCookies(name);
+}
+
+async function getFromCookies(name: string) {
+    let value = "; " + (await CookieHandlerReference.getReferenceOrThrow().cookieHandler.getCookie());
+    let parts = value.split("; " + name + "=");
+    if (parts.length >= 2) {
+        let last = parts.pop();
+        if (last !== undefined) {
+            return last.split(";").shift();
         }
     }
+    return undefined;
+}
 
-    const { status } = await getIdRefreshToken(false);
+async function setAuthorizationHeaderIfRequired(clonedHeaders: Headers, addRefreshToken: boolean = false) {
+    logDebugMessage("setTokenHeaders: adding existing tokens as header");
 
-    logDebugMessage("setIdRefreshToken: setting: " + idRefreshToken);
-    await setIDToCookie(idRefreshToken, AuthHttpRequest.config.sessionScope);
+    // We set the Authorization header even if the tokenTransferMethod preference set in the config is cookies
+    // since the active session may be using cookies. By default, we want to allow users to continue these sessions.
+    // The new session preference should be applied at the start of the next session, if the backend allows it.
 
-    if (idRefreshToken === "remove" && status === "EXISTS") {
-        // we check for wasLoggedIn cause we don't want to fire an event
-        // unnecessarily on first app load or if the user tried
-        // to query an API that returned 401 while the user was not logged in...
-        if (statusCode === AuthHttpRequest.config.sessionExpiredStatusCode) {
-            logDebugMessage("setIdRefreshToken: firing UNAUTHORISED event");
-            AuthHttpRequest.config.onHandleEvent({
-                action: "UNAUTHORISED",
-                sessionExpiredOrRevoked: true,
-                userContext: {}
-            });
+    const accessToken = await getTokenForHeaderAuth("access");
+    const refreshToken = await getTokenForHeaderAuth("refresh");
+
+    // We don't add the refresh token because that's only required by the refresh call which is done with fetch
+    // Still, we only add the Authorization header if both are present, because we are planning to add an option to expose the
+    // access token to the frontend while using cookie based auth - so that users can get the access token to use
+    if (accessToken !== undefined && refreshToken !== undefined) {
+        // the Headers class normalizes header names so we don't have to worry about casing
+        if (clonedHeaders.has("Authorization")) {
```

Comment:
what if the user uses lowercase authorization?

Reply:
the Headers class handles these as case insensitive.